### ISSUE TITLE
[bug] game closed

### ISSUE THREAD

**Issue by lublak** [dannydjdk/Tiny-Redstone#17]
I made it :P i try to recreate it, my test world cannot be opened.
Currently I have no idea how, but it had something to do with pistons and 2 panels.

```
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 20.03.21 18:08
Description: Ticking block entity

java.lang.NullPointerException: Ticking block entity
	at com.dannyandson.tinyredstone.blocks.PanelTile.getNeighborTileCellIndex(PanelTile.java:804) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborTileCell(PanelTile.java:797) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighbor(PanelTile.java:655) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborCells(PanelTile.java:631) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborCells(PanelTile.java:624) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updatePiston(PanelTile.java:412) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.func_73660_a(PanelTile.java:318) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at net.minecraft.world.World.func_217391_K(World.java:1279) ~[?:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:performant.mixins.json:world.WorldMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:371) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:performant.mixins.json:world.ServerWorldBlockUpdateMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:APP:quark.mixins.json:ServerWorldMixin,pl:mixin:APP:charm.mixins.json:ServerWorldMixin,pl:mixin:APP:charm.mixins.json:accessor.ServerWorldAccessor,pl:mixin:APP:performant.mixins.json:world.ServerWorldMixin,pl:mixin:APP:performant.mixins.json:entity.MobServerWorldMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:851) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:charm.mixins.json:accessor.MinecraftServerAccessor,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:787) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:charm.mixins.json:accessor.MinecraftServerAccessor,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.integrated.IntegratedServer.func_71217_p(IntegratedServer.java:118) ~[?:?] {re:mixin,re:classloading,xf:OptiFine:default,pl:mixin:APP:smoothboot.mixins.json:client.IntegratedServerMixin,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.func_240802_v_(MinecraftServer.java:642) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:charm.mixins.json:accessor.MinecraftServerAccessor,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:A}
	at net.minecraft.server.MinecraftServer.lambda$startServer$0(MinecraftServer.java:232) ~[?:?] {re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:structure_gel.mixins.json:MinecraftServerMixin,pl:mixin:APP:charm.mixins.json:accessor.MinecraftServerAccessor,pl:mixin:APP:byg.mixins.json:server.MixinMinecraftServer,pl:mixin:A}
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_242] {}


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at com.dannyandson.tinyredstone.blocks.PanelTile.getNeighborTileCellIndex(PanelTile.java:804) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborTileCell(PanelTile.java:797) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighbor(PanelTile.java:655) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborCells(PanelTile.java:631) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updateNeighborCells(PanelTile.java:624) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.updatePiston(PanelTile.java:412) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
	at com.dannyandson.tinyredstone.blocks.PanelTile.func_73660_a(PanelTile.java:318) ~[tinyredstone:1.16.5-1.5.0b] {re:classloading}
-- Block entity being ticked --
Details:
	Name: tinyredstone:redstone_panel // com.dannyandson.tinyredstone.blocks.PanelTile
	Block: Block{tinyredstone:redstone_panel}[facing=down]
	Block location: World: (-81,71,28), Chunk: (at 15,4,12 in -6,1; contains blocks -96,0,16 to -81,255,31), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
	Block: Block{tinyredstone:redstone_panel}[facing=down]
	Block location: World: (-81,71,28), Chunk: (at 15,4,12 in -6,1; contains blocks -96,0,16 to -81,255,31), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,0,0 to -1,255,511)
Stacktrace:
	at net.minecraft.world.World.func_217391_K(World.java:1279) ~[?:?] {re:computing_frames,pl:accesstransformer:B,re:mixin,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:performant.mixins.json:world.WorldMixin,pl:mixin:A}
	at net.minecraft.world.server.ServerWorld.func_72835_b(ServerWorld.java:371) ~[?:?] {re:mixin,pl:accesstransformer:B,re:computing_frames,pl:accesstransformer:B,re:classloading,pl:accesstransformer:B,pl:mixin:APP:performant.mixins.json:world.ServerWorldBlockUpdateMixin,pl:mixin:APP:abnormals_core.mixins.json:ServerWorldMixin,pl:mixin:APP:endergetic.mixins.json:ServerWorldMixin,pl:mixin:APP:quark.mixins.json:ServerWorldMixin,pl:mixin:APP:charm.mixins.json:ServerWorldMixin,pl:mixin:APP:charm.mixins.json:accessor.ServerWorldAccessor,pl:mixin:APP:performant.mixins.json:world.ServerWorldMixin,pl:mixin:APP:performant.mixins.json:entity.MobServerWorldMixin,pl:mixin:A}
```

**Comment by dannydjdk** [comment]
Ooh! The first public crash.
If you want to get back in that world, open the forge-server.toml file inside the world folder->serverconfig folder and change "removeErroringTileEntities = false" to "removeErroringTileEntities = true". That should just disable that block.
Obviously, that's just a temporary workaround until I fix this.

Can you give me any details on what you were doing before that happened?

**Comment by dannydjdk** [comment]
Actually, that bug may already be fixed in my most recent update. It looks like it's related to #16 .

**Comment by lublak** [comment]
@dannydjdk
Unfortunately, I have no idea until now in which combination I had the tiny redstone.
I have already recreated the world. It was just a test world.
When I have something, I will write something about it.
But I will wait for #16 first.

**Comment by dannydjdk** [comment]
Ok. I just reproduced it. Definitely not fixed with the last push. Should be fixed soon though.

**Comment by lublak** [comment]
@dannydjdk What have I done? I am interested.

**Comment by dannydjdk** [comment]
For me, it happened when I extended a piston on to an adjacent tile where there was no block to push. When the piston retracted, it crashed. So, one panel was telling the other panel to move a block back toward the first panel.

**Comment by dannydjdk** [comment]
Released with 1.16.5-1.5.1b
